src/query/expression/src/types.rs

```diff
@@ -113,6 +113,9 @@ pub trait ValueType: Debug + Clone + PartialEq + Sized + 'static {
     type ColumnIterator<'a>: Iterator<Item = Self::ScalarRef<'a>> + TrustedLen;
     type ColumnBuilder: Debug + Clone + PartialEq;
 
+    /// Upcast GAT type's lifetime.
+    fn upcast_gat<'short, 'long: 'short>(long: Self::ScalarRef<'long>) -> Self::ScalarRef<'short>;
```

Comment:
I think it's useless..

Reply:
It's used for such codes:

```rust
#[derive(Default)]
pub struct CmpMin;

impl<T> ChangeIf<T> for CmpMin
where
    T: ValueType,
    for<'a> T::ScalarRef<'a>: PartialOrd,
{
    #[inline]
    fn change_if<'a>(l: T::ScalarRef<'_>, r: T::ScalarRef<'_>) -> bool {
        let l = T::upcast_gat(l);
        let r = T::upcast_gat(r);
        l.partial_cmp(&r).unwrap_or(Ordering::Equal) == Ordering::Greater
    }
}
```

Or there will be lifetime problems